File: Daily Test/Java/Sort By BitScore 6.10.2025.java
import java.util.*;

class NumCalc {
    int val, calc;
    NumCalc(int v, int c) { val = v; calc = c; }
}

public class Main {
    static int binVal(int n) {
        int b = 0, m = 1;
        while (n > 0) {
            b += (n % 2) * m;
            m *= 10;
            n /= 2;
        }
        return b;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        NumCalc[] arr = new NumCalc[n];
        for (int i = 0; i < n; i++) {
            String s = sc.next();
            int val = Integer.parseInt(s), calc = 0;
            for (char ch : s.toCharArray())
                calc += (ch - '0') * binVal(ch - '0');
            arr[i] = new NumCalc(val, calc);
        }
        Arrays.sort(arr, (a, b) -> a.calc == b.calc ? a.val - b.val : a.calc - b.calc);
        for (NumCalc x : arr) System.out.print(x.val + " ");
    }
}

File: Daily Test/C/Sort By BitScore 6.10.2025.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    int val;
    int calc;
} NumCalc;

int bin_val(int n) {
    int b = 0, m = 1;
    while (n) {
        b += (n % 2) * m;
        m *= 10;
        n /= 2;
    }
    return b;
}

int compare(const void* a, const void* b) {
    NumCalc* x = (NumCalc*)a;
    NumCalc* y = (NumCalc*)b;
    if (x->calc == y->calc) return x->val - y->val;
    return x->calc - y->calc;
}

int main() {
    int n;
    scanf("%d", &n);
    NumCalc arr[n];
    char buf[32];
    for (int i = 0; i < n; i++) {
        scanf("%s", buf);
        arr[i].val = atoi(buf);
        arr[i].calc = 0;
        for (int j = 0; buf[j]; j++)
            arr[i].calc += (buf[j] - '0') * bin_val(buf[j] - '0');
    }
    qsort(arr, n, sizeof(NumCalc), compare);
    for (int i = 0; i < n; i++) printf("%d ", arr[i].val);
    return 0;
}